gems/smithy/lib/smithy/model.rb

```diff
@@ -9,28 +9,38 @@
 module Smithy
   # A module that parses the Smithy JSON model.
   module Model
-    PRELUDE_SHAPES = %w[
-      smithy.api#BigInteger
-      smithy.api#BigDecimal
-      smithy.api#Blob
-      smithy.api#Boolean
-      smithy.api#Byte
-      smithy.api#Document
-      smithy.api#Double
-      smithy.api#Float
-      smithy.api#Integer
-      smithy.api#Long
-      smithy.api#PrimitiveBoolean
-      smithy.api#PrimitiveByte
-      smithy.api#PrimitiveDouble
-      smithy.api#PrimitiveFloat
-      smithy.api#PrimitiveInteger
-      smithy.api#PrimitiveLong
-      smithy.api#PrimitiveShort
-      smithy.api#Short
-      smithy.api#String
-      smithy.api#Timestamp
-      smithy.api#Unit
-    ].freeze
+    PRELUDE_SHAPES = {
+      'smithy.api#BigInteger' => { 'type' => 'bigInteger' },
+      'smithy.api#BigDecimal' => { 'type' => 'bigDecimal' },
+      'smithy.api#Blob' => { 'type' => 'blob' },
+      'smithy.api#Boolean' => { 'type' => 'boolean' },
+      'smithy.api#Byte' => { 'type' => 'byte' },
+      'smithy.api#Document' => { 'type' => 'document' },
+      'smithy.api#Double' => { 'type' => 'double' },
+      'smithy.api#Float' => { 'type' => 'float' },
+      'smithy.api#Integer' => { 'type' => 'integer' },
+      'smithy.api#Long' => { 'type' => 'long' },
+      'smithy.api#PrimitiveBoolean' => { 'type' => 'boolean', 'traits' => { 'smithy.api#default' => false } },
+      'smithy.api#PrimitiveByte' => { 'type' => 'byte', 'traits' => { 'smithy.api#default' => 0 } },
+      'smithy.api#PrimitiveDouble' => { 'type' => 'double', 'traits' => { 'smithy.api#default' => 0 } },
+      'smithy.api#PrimitiveFloat' => { 'type' => 'float', 'traits' => { 'smithy.api#default' => 0 } },
+      'smithy.api#PrimitiveInteger' => { 'type' => 'integer', 'traits' => { 'smithy.api#default' => 0 } },
+      'smithy.api#PrimitiveLong' => { 'type' => 'long', 'traits' => { 'smithy.api#default' => 0 } },
+      'smithy.api#PrimitiveShort' => { 'type' => 'short', 'traits' => { 'smithy.api#default' => 0 } },
+      'smithy.api#Short' => { 'type' => 'short' },
+      'smithy.api#String' => { 'type' => 'string' },
+      'smithy.api#Timestamp' => { 'type' => 'timestamp' },
+      'smithy.api#Unit' => { 'type' => 'structure', 'members' => {}, 'traits' => { 'smithy.api#unitType' => {} } }
+    }.freeze
+
+    def self.shape(model, target)
```

Comment:
I'm finding this a little weird - I think its a little weird to have a module `shape` method that takes a model rather than a model class with a `shape` method.  It seems like we want a class for model instead of just a Hash.

Reply:
It would be an instance that we would have to create in many places and only provide a value. It doesn't require any state. This is similar to Api.shape in v3 code generation.

Reply:
This to me is still a code smell that using Hash is the wrong way to represent model if we then need static methods to actually use it.

But I'll disagree and commit on it.

Reply:
We tried wrapping the model with Vise and it didn't work out too well from the preprocess standpoint. Even if the model was wrapped, trying to fetch the shape would be the same result, either it would be another hash/object or failure.